# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: 0b4595b2c153
# Full Hash: 0b4595b2c1531b83fc24fbf9b0f6ceacfc3506bd
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 09:35:42
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 4363e3a3d799 -r 0b4595b2c153 layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Wed Apr 22 01:37:52 2020 +0000
+++ b/layout/printing/nsPrintJob.cpp	Wed Apr 22 01:37:55 2020 +0000
@@ -3195,7 +3195,7 @@
     // dump the views of the sub documents
     int32_t i, n;
     BrowsingContext* bc = nsDocShell::Cast(aDocShell)->GetBrowsingContext();
-    for (auto& child : bc->GetChildren()) {
+    for (auto& child : bc->Children()) {
       if (auto childDS = child->GetDocShell()) {
         DumpViews(childAsShell, out);
       }